### DISCUSSION TITLE
I'd like to add a button to the text field autoform and another button to the image field autoform

### DISCUSSION THREAD

**Discussion by pgibler** [react-page/react-page#1288] [Ideas] [answered]
I'm wondering how I could this.
There are 2 approaches I can think of:
1. Somehow augmenting the existing text & image autoforms to include my new buttons.
2. Creating new custom plugins that re-implement the text and image plugins and include the new buttons
Which approach should I use here? I'd think #2 would be a good way to do it if I can compose the new elements. #1 would be easiest if that's an option.

**Comment by macrozone** [comment]
@pgibler can you describe the use case?

**Reply by pgibler** [reply]
My use case is to allow a user to generate the text for the selected text input by invoking a service that calls an LLM (i.e. GPT-3.5). I think that adding a button to the autoform to invoke this service would be a good approach. I'm open to suggestion, of course.

Ideally I would like to compose a custom text plugin using the existing slate plugin parts and adding my service invoke button via parameter.

**Reply by pgibler** [reply, marked as answer]
@macrozone Read through the documentation again and came upon this decoration system:

https://react-page.github.io/docs/#/recipes?id=add-additional-controls-to-all-cell-plugins
https://react-page.github.io/examples/decorateplugins

This appears to be what I need. I'll close this issue now and begin reading / using the decoration system for now. If you have anything to mention, let me know. Thanks.
